ConcurrentModificationException in EventLoops.stopAll

Just caught it when running `software.chronicle.enterprise.map.ConcurrentReceiversTest`

```
java.util.concurrent.ExecutionException: java.util.ConcurrentModificationException: java.util.ConcurrentModificationException
	at java.util.concurrent.ForkJoinTask.get(ForkJoinTask.java:1006)
	at net.openhft.chronicle.threads.EventLoops.stopAll(EventLoops.java:29)
	at net.openhft.chronicle.network.cluster.ClusterContext.performStop(ClusterContext.java:383)
	at net.openhft.chronicle.network.cluster.ClusterContext.close(ClusterContext.java:363)
	at net.openhft.chronicle.core.io.Closeable.closeQuietly(Closeable.java:45)
	at net.openhft.chronicle.core.io.Closeable.closeQuietly(Closeable.java:33)
	at net.openhft.chronicle.network.cluster.Cluster.performClose(Cluster.java:109)
	at net.openhft.chronicle.core.io.SimpleCloseable.close(SimpleCloseable.java:36)
	at net.openhft.chronicle.core.io.Closeable.closeQuietly(Closeable.java:45)
	at net.openhft.chronicle.core.io.Closeable.closeQuietly(Closeable.java:33)
	at software.chronicle.enterprise.map.ReplicatedMap.performClose(ReplicatedMap.java:246)
	at net.openhft.chronicle.core.io.AbstractCloseable.callPerformClose(AbstractCloseable.java:326)
	at net.openhft.chronicle.core.io.AbstractCloseable.close(AbstractCloseable.java:261)
	at software.chronicle.enterprise.map.ConcurrentReceiversTest.run(ConcurrentReceiversTest.java:95)
	at software.chronicle.enterprise.map.ConcurrentReceiversTest.testTight(ConcurrentReceiversTest.java:48)
	at sun.reflect.GeneratedMethodAccessor6.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:52)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)
Caused by: java.util.ConcurrentModificationException: java.util.ConcurrentModificationException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:593)
	at java.util.concurrent.ForkJoinTask.get(ForkJoinTask.java:1005)
	... 40 more
Caused by: java.util.ConcurrentModificationException
	at java.util.ArrayList.forEach(ArrayList.java:1262)
	at net.openhft.chronicle.threads.MonitorEventLoop.performStop(MonitorEventLoop.java:79)
	at net.openhft.chronicle.threads.MonitorEventLoop.performStopFromStarted(MonitorEventLoop.java:73)
	at net.openhft.chronicle.threads.AbstractLifecycleEventLoop.stop(AbstractLifecycleEventLoop.java:56)
	at net.openhft.chronicle.threads.EventGroup.performStop(EventGroup.java:329)
	at net.openhft.chronicle.threads.EventGroup.performStopFromStarted(EventGroup.java:325)
	at net.openhft.chronicle.threads.AbstractLifecycleEventLoop.stop(AbstractLifecycleEventLoop.java:56)
	at net.openhft.chronicle.threads.EventLoops.lambda$addAllEventLoopStoppers$0(EventLoops.java:46)
	at java.util.concurrent.ForkJoinTask$AdaptedCallable.exec(ForkJoinTask.java:1424)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:175)

```

--- a/src/main/java/net/openhft/chronicle/threads/MonitorEventLoop.java
+++ b/src/main/java/net/openhft/chronicle/threads/MonitorEventLoop.java
@@ -76,7 +76,10 @@ public class MonitorEventLoop extends AbstractLifecycleEventLoop implements Runn
     private void performStop() {
         unpause();
         Threads.shutdownDaemon(service);
-        handlers.forEach(Threads::loopFinishedQuietly);
+
+        synchronized (handlers) {
+            handlers.forEach(Threads::loopFinishedQuietly);
+        }
     }
 
     @Override
@@ -159,7 +162,10 @@ public class MonitorEventLoop extends AbstractLifecycleEventLoop implements Runn
     @Override
     protected void performClose() {
         super.performClose();
-        net.openhft.chronicle.core.io.Closeable.closeQuietly(handlers);
+
+        synchronized (handlers) {
+            net.openhft.chronicle.core.io.Closeable.closeQuietly(handlers);
+        }
     }
 
     /**

